### ISSUE TITLE
Filter resolvers to compatible address type instead of trying resolvers until a compatible address type comes out

### ISSUE THREAD

**Issue by jcgruenhage** [avitex/rust-public-ip#22]

**Comment by aWeinzierl** [comment]
Is this really the case?
The family is passed to the resolver, see: https://github.com/avitex/rust-public-ip/blob/675505fd52d56805a3684004ad5d0a6523def2b7/src/lib.rs#L197

It is not expected that a different family is returned, but checked still.

**Comment by jcgruenhage** [comment]
Yes, to my understanding of this crate, this is the case. The resolver does get the IP, but that doesn't mean that it only tries resolvers that are compatible. For example, the api.ipify.org http resolver doesn't support IPv6:

```
~ 
❯ dig +short A api.ipify.org
104.26.12.205
104.26.13.205
172.67.74.152

~ 
❯ dig +short AAAA api.ipify.org

~ 
❯ 
```

The resolver will still try to resolve a v6 address for api.ipify.org, fail, and return an error. For other resolvers that are dual-stack capable, they will successfully resolve a v6 address even when trying to get a v4 address, use that to obtain the public v6 address, only to throw it away afterwards.
